DEHPEcosimPro/ViewModel/Dialogs/HubMappingConfigurationDialogViewModel.cs

```diff
@@ -438,21 +576,213 @@ private void UpdateProperties()
+                .Where(x => this.DstController.IsVariableWritable(x.Reference)
+                            && this.GetNameFromArrayReference(x.Reference) == variableName).ToList();
+
+            var listOfVariableRowViewModel = listOfValueByRowsAndColumns.Select(x =>
+            {
+                var (reference, _) = x;
+                return new VariableRowViewModel((reference, this.DstController.ReadNode(x.Reference)));
+            }).ToList();
+            for (int i = 0; i < listOfVariableRowViewModel.Count(); i++)
+            {
+                foreach (var parameterChoosen in this.ListOfRowsToMap.ListOfVariableToMap)
+                {
+                    this.GetRowAndColumnNumberFromName(listOfVariableRowViewModel[i].Name, out int rowNumber, out int colNumber);
+                    
+                    if (parameterChoosen.SelectedColumnMatched != null && (colNumber == 0 || parameterChoosen.Index.Contains(colNumber.ToString())))
                     {
-                        r.Node = this.DstController.ReadNode(r.Reference);
-                        return new VariableRowViewModel(r, false)
+                        var columnOfParameter = parameterChoosen.SelectedColumnMatched;
+                        var colData = tableListOfTuple.FirstOrDefault(x => x.name == columnOfParameter);
+
+                        var variableRowViewModel = listOfVariableRowViewModel.FirstOrDefault(x => x.Name == listOfVariableRowViewModel[i].Name);
+
+                        var element = new MappedElementDefinitionRowViewModel()
                         {
-                            HasWriteAccess = true
+                            SelectedParameter = parameter,
+                            SelectedValue = new ValueSetValueRowViewModel(parameter.QueryParameterBaseValueSet(this.SelectedOption, this.SelectedState), colData.list[rowNumber - 1], null),
+                            SelectedVariable = variableRowViewModel,
                         };
-                    }));
 
-            this.MappedElements.ChangeTrackingEnabled = true;
+                        element.VerifyValidity();
+
+                        if (element.IsValid)
+                        {
+                            lisOfMappedElement.Add(element);
+                            this.MappedElements.Add(element);
+                        }
+
+                    }
+                    var result = lisOfMappedElement.Select(x => x.SelectedVariable.Name).ToList();
+                }
+
+            }
+        }
+
+        /// <summary>
+        /// Get as a number the row and the colomn out of the name
+        /// </summary>
+        /// <param name="name">name of a value, should include [x] or [x,y]</param>
+        /// <param name="rowNumber">the row of the value</param>
+        /// <param name="colNumber">the column of the value</param>
+        private void GetRowAndColumnNumberFromName(string name, out int rowNumber, out int colNumber)
```

Comment:
It looks like there was another method above that was doing the same thing?

Either way, same question as before, is there no better way to find dimensions than from the string?